cmd/distributor/main.go

```diff
@@ -48,6 +48,8 @@ func main() {
 	if err != nil {
 		log.Fatalf("Error initializing distributor: %v", err)
 	}
+	go dist.Run()
```

Comment:
I was discussing this with @jml yesterday: in general, I prefer if goroutines are launched inside the constructor, mainly because these components are used in multiple places and its a pain to have to repeat the code everywhere (for instance, you didn't do `go run dist.Run()` in the ruler or the querieir`).  WDYT?